content/blog/colour-scales.md

```diff
@@ -0,0 +1,70 @@
+---
+author: "Emma Russell"
+date: 2020-05-30
+title: "Shades of meaning: visualising geographical data with colour in a data-driven web application"
+best: false
+tags:
+ - visualisation
+ - UI
+---
+
+Colour and maps are two crucial aspects of data visualisation, and putting them together is a classic technique for revealing geographical patterns in data. 
+
+[{{< figure src="/img/cscale_smith_map.jpg" title="William Smith's geological map of 1815" >}}](https://www.nhm.ac.uk/discover/first-geological-map-of-britain.html)
+ 
+If data values are discrete, selecting the colours to use is straightforward - just choose an appropriate colour for each value. For continuous data, a colour scale is required, where any numeric value in the data range can be mapped to a colour, and where the gradations in colour are visually meaningful. (This type of map is known as a [choropleth](https://en.wikipedia.org/wiki/Choropleth_map).)
+
+This is fine for a pre-canned visualisation which can be tailored to the data, but trickier in dynamic applications with data whose extents are not known in advance. Pre-defining a colour scale which will be appropriate for all possible datasets can be problematic, as we discovered when we developed the web application part of the [Naomi](https://reside-ic.github.io/projects/naomi/) project for estimating district-level HIV indicators in sub-Saharan Africa.
+
+A key visualisation feature in this application are maps whose regions are coloured to show indicator values as shades on a colour scale. The application is designed so that metadata defining data formats and presentation originate in the research code. This metadata includes the colour scales which the maps use to display regional data.
+
+{{< figure src="/img/cscale_architecture.png" title="Naomi architecture" >}}
+
+This approach allows for a flexible, iterative workflow where scientists developing research code can experiment with new data series, model parameters and visualisation settings without requiring further development from the web application team. 
```

Comment:
This is supposed to explain the convenience of the metadata-based architecture so that's why I thought it was worth mentioning the model run parameters.

Reply:
I think mentioning model params is just confusing here because they aren't relevant to this bit of the app.